specs-go/config.go

```diff
@@ -141,7 +141,7 @@ type Linux struct {
 	// If resources are specified, the cgroups at CgroupsPath will be updated based on resources.
 	CgroupsPath *string `json:"cgroupsPath,omitempty"`
 	// Namespaces contains the namespaces that are created and/or joined by the container
-	Namespaces []LinuxNamespace `json:"namespaces,omitempty"`
+	Namespaces map[string]LinuxNamespace `json:"namespaces,omitempty"`
```

Comment:
It should be `map[LinuxNamespaceType]LinuxNamespace`, what's more, maybe we need to rename `LinuxNamespace` to `LinuxNamespaceProperty`